docs/zkapps/advanced-snarkyjs/time-locked-accounts.mdx

```diff
@@ -0,0 +1,121 @@
+    src="/img/10_time_locked_accounts.png"
+    width="95%"
+    alt="Timing parameters"
+  />
+</figure>
+
+The red cross on the left marks the point in time where the `timing` field is set and `isTimed` switches from `false` to `true`. The blue line shows how the amount of unlocked tokens increases over time, until it finally reaches its maximum value and stays flat. At this point, `isTimed` flips from `true` back to `false`, because no tokens remain locked.
+
+As you can see, the maximum amount of unlocked tokens is defined by the `initialMinimumBalance`. It is called "initialMinimumBalance" because, even though the tokens show up in the balance, they can't be withdrawn — in other words, the account has a a non-zero _minimum balance_. Initially, that minimum balance is equal to the amount of tokens locked (i.e., the initial minimum balance). Over time, the minimum balance decreases until it hits zero, which is the condition that makes `isTimed` false again.
+
+The other timing-related properties are described as follows:
+
+- `cliffTime`: the initial time period during which all tokens are locked. Note that 'time' is measured in Mina by 'slots', where 1 slot is 3min currently.
+- `cliffAmount`: the quantity of tokens to be unlocked when the cliff time has elapsed. If this is greater or equal the 'initial minimum balance', all tokens will be unlocked after the cliff time elapses.
+- `vestingPeriod`: After the cliff time elapses, tokens can be set to unlock periodically at a fixed interval, by a fixed quantity. The vesting period is the length of that interval.
+- `vestingIncrement`: the quantity of tokens unlocked after each vesting period elapses.
+
+:::note
+Only one vesting schedule can be specified per account and it cannot be changed during the vesting period.
+
+Because of this, when `isTimed` is set to `true`, the values of the timing fields are immutable and cannot be changed.
+After all tokens are unlocked and `isTimed` flips back to `false`, the account timing will become mutable again.
+:::
+
+### Setting timing in SnarkyJS
+
+In SnarkyJS, `timing` is one of the account fields that can be updated by using an account update:
+
+```ts
+accountUpdate.account.timing.set({ initialMinimumBalance, cliffTime, ...etc });
+```
+
+When setting this, all timing-related properties are required, except for `isTimed`, which will be set the protocol automatically.
+
+### Examples
+
+Let's see how to correctly implement some simple example use cases.
+
+#### Example 1: All tokens unlock after 1 week
+
+If you simply want all tokens to unlock after a certain time, then the only properties you need to care about are `cliffTime` and `cliffAmount`. We also want to set `initialMinimumBalance` equal to `cliffAmount` to ensure all tokens are unlocked when the cliff elapses. Both `vestingPeriod` and `vestingIncrement` are unused so we'll set them to their default values, 1 and 0:
+
+```ts
+// example: 10 locked MINA
+const totalAmount = UInt64.from(10e9);
```

Comment:
Realized this would be more clear:
`const totalAmount = UInt64.from(10e9);`
as 
`const tokensToLock = UInt64.from(10e9);`
no? 

(And updates to subsequent lines and in example 2 also.)

Suggestion because it makes it clear that they can choose to lock a portion of the tokens, which is true, correct?

Reply:
yes, the account can be used as normal account with tokens in addition to the locked ones that can go in and out normally